### ISSUE TITLE
Modules invoked with missing required parameters

### ISSUE THREAD

**Issue by mromascanu123** [terraform-google-modules/terraform-example-foundation#1186]
### TL;DR

When invoking some library modules required parameters not passed. It might work if the declared parameters (in variables.tf of the module) endup by actually not being used for reasons of "example" code or disabled sections. But not sure what is the general opinion regarding this coding practice, which I would abstain to qualify in my own words.

See as an example (among many) the attached screenshot exemplifying a call to project-factory with missing required parameters in 1-org/envs/shared/projects.tf. 

### Expected behavior

Required parameters with valid values should be passed to the modules requiring them, regardless whether actually being used in a particular instance of a module

### Observed behavior

The deployment "seems" to work although with many warnings - can't tell anything about actual vs expected resources being deployed

### Terraform Configuration

```hcl
See screenshot as an example, it is a matter of code not configuration
```

### Terraform Version

```sh
terraform version
Terraform v1.6.0
on linux_amd64

Your version of Terraform is out of date! The latest version
is 1.8.0.
```

### Additional information

![missing-params](https://github.com/terraform-google-modules/terraform-example-foundation/assets/162719251/c55b7d39-11a7-4279-8aa7-de115b69492c)

**Comment by daniel-cit** [comment]
Thanks for your report of this issue.

The **Terraform configuration is valid** as validated by `terraform validate`

![image](https://github.com/terraform-google-modules/terraform-example-foundation/assets/55760939/336f8678-4657-4b29-968d-4fcb5391bcc1)

Regarding the list of missing required variables in the screenshot most of then are not part of the project-factory module [variable list](https://github.com/terraform-google-modules/terraform-google-project-factory/blob/v14.0.0/variables.tf) like `admin_email`, `amount`, and `api_sa_group` among others.

For the ones that are part of the module like `domain`, they **do have a default value**

https://github.com/terraform-google-modules/terraform-google-project-factory/blob/v14.0.0/variables.tf#L34C1-L38C2
![image](https://github.com/terraform-google-modules/terraform-example-foundation/assets/55760939/4d7b8009-aeab-4042-a1f9-1d49cdf9c922)

The official Terraform documentation regarding [input variables](https://developer.hashicorp.com/terraform/language/values/variables#arguments), states that variables **with a default value are optional (not required)**

![image](https://github.com/terraform-google-modules/terraform-example-foundation/assets/55760939/37e1e7ef-05a8-4b04-a25f-8a29c0cc455e)

![image](https://github.com/terraform-google-modules/terraform-example-foundation/assets/55760939/fc9fe0ed-2e3f-44d3-aaf8-8e8bd3e7dc84)

Reported behavior looks like an issue in the the configuration of the VSCode extension for Terraform,
possibly related to this issue [fix: Work around 'unreliable' input data for Registry modules](https://github.com/hashicorp/terraform-ls/pull/1456)
or a similar one in the `terraform-ls`  extension for VSCode provided by HashiCorp since running `terraform validate` proves that the configuration is valid.

Updating the VSCode extension for Terraform to use the latest release should be able to solve this issue.

**Comment by eeaton** [comment]
Thanks Daniel, I agree with this explanation that it looks like a configuration issue on the IDE and not a bug in this repo's code. I'll close this issue, please re-open if there are further concerns.
